src/frontend/toolbar-button/TeamInfo/index.tsx

```diff
@@ -158,30 +162,50 @@ async function openNewPontoonTabAndClosePopup(url: string): Promise<void> {
 export const TeamInfo: React.FC = () => {
   const [projectForCurrentTab, setProjectForCurrentTab] =
     useState<StorageContent['projectsList'][string]>();
-  const [team, setTeam] = useState<StorageContent['teamsList'][string]>();
+  const [team, setTeam] = useState<StorageContent['team']>();
   const [teamActivity, setTeamActivity] =
     useState<StorageContent['latestTeamsActivity'][string]>();
   const [pontoonBaseUrl, setPontoonBaseUrl] =
     useState<OptionsContent['pontoon_base_url']>();
+  const [teamCode, setTeamCode] = useState<OptionsContent['locale_team']>();
 
+  // Initial load
   useEffect(() => {
     (async () => {
-      const [
-        projectForCurrentTab,
-        { teamsList, latestTeamsActivity },
-        { locale_team: teamCode, pontoon_base_url },
-      ] = await Promise.all([
-        getPontoonProjectForTheCurrentTab(),
-        getFromStorage(['teamsList', 'latestTeamsActivity']),
-        getOptions(['locale_team', 'pontoon_base_url']),
+      const { locale_team, pontoon_base_url } = await getOptions([
+        'locale_team',
+        'pontoon_base_url',
       ]);
+      setTeamCode(locale_team);
+      setPontoonBaseUrl(pontoon_base_url);
+    })();
+  }, []);
+
+  // Load team data via teamCode change
+  useEffect(() => {
+    if (!teamCode) return;
+
+    (async () => {
+      const [projectForCurrentTab, { team, latestTeamsActivity }] =
+        await Promise.all([
+          getPontoonProjectForTheCurrentTab(),
+          getFromStorage(['team', 'latestTeamsActivity']),
+        ]);
       setProjectForCurrentTab(projectForCurrentTab);
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      setTeam(teamsList![teamCode]);
+      setTeam(team!);
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       setTeamActivity(latestTeamsActivity![teamCode]);
-      setPontoonBaseUrl(pontoon_base_url);
     })();
+  }, [teamCode]);
+
+  // locale_team option listener
+  useEffect(() => {
+    const unsubscribe = listenToOptionChange('locale_team', async () => {
+      const teamCode = await getOneOption('locale_team');
+      setTeamCode(teamCode);
+    });
+    return unsubscribe;
   }, []);
```

Comment:
Prettier will likely reformat this, but something like this ought to be simpler.

Also, the comment here is rather unnecessary.

```suggestion
  useEffect(() => listenToOptionChange(
    'locale_team',
    () => getOneOption('locale_team').then(setTeamCode)
  ), []);
```